src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -1297,6 +1297,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return typeSymbol.IsWellKnownCompilerServicesTopLevelType("IsExternalInit")
         End Function
 
+        <Extension>
+        Friend Function IsWellKnownTypeLock(typeSymbol As TypeSymbol) As Boolean
```

Comment:
It looks like this function doesn't have much test coverage, especially for negative results.